pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/blocks/BlockSplitter.java

```diff
@@ -0,0 +1,36 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.query.runtime.blocks;
+
+import java.util.Iterator;
+import org.apache.pinot.common.datablock.BaseDataBlock;
+
+
+/**
+ * Interface for splitting transferable blocks. This is used for ensuring
+ * that the blocks that are sent along the wire play nicely with the
+ * underlying transport.
+ */
+public interface BlockSplitter {
```

Comment:
seems like we can also make this an inner interface in BlockExchange.java? 

any chance we will use this other than test purpose?

Reply:
> seems like we can also make this an inner interface in BlockExchange.java?

Yeah we can do that, I left it outside because I thought there might actually be some pretty good reasons to have multiple types of block splitters (e.g. for columnar once we support that).

> any chance we will use this other than test purpose?

It already is - `MailboxSendOperator` already uses it to abstract away the difference between `isLeafStageSedner` and not. See:
```
    BlockSplitter splitter = (block, type, size)
         -> TransferableBlockUtils.splitBlock(block, type, size, isLeafStageSender);
     _exchange = blockExchangeFactory.build(mailboxService, receivingMailboxes, exchangeType, keySelector, splitter);
```